docs(storage): add samples for lifer cycle and versioning

HemangChothani opened the pull request:
Added Samples For

- Enable, Disable Versioning

- Enable.Disable life cycle  management

- Set blob metadata

HemangChothani committed:
docs(storage): add samples for lifer cycle and versioning

mf2199 commented on storage/cloud-client/snippets_test.py:
```diff
@@ -156,6 +161,12 @@ def test_blob_metadata(test_blob, capsys):
     assert test_blob.name in out
 
 
+def test_set_blob_metadata(test_blob, capsys):
+    storage_set_metadata.set_blob_metadata(test_blob.bucket.name, test_blob.name)
+    out, _ = capsys.readouterr()
+    assert test_blob.name in out
```
Merely a question, is there a reason to putting this method before `test_delete_blob`, separately from the other new ones?

HemangChothani replied:
No, just put because above method is get_blob_metadata so pu set_metadata here.

mf2199 commented on storage/cloud-client/snippets_test.py:
```diff
@@ -249,3 +260,30 @@ def test_copy_blob(test_blob):
+def test_versioning(test_bucket, capsys):
+    bucket = storage_enable_versioning.enable_versioning(test_bucket)
+    out, _ = capsys.readouterr()
+    assert "Versioning was enabled for bucket" in out
```
This will fail if the definition of the error string will change.

HemangChothani replied:
Yes, but same implemented in bigquery sample as well as you can see here https://github.com/googleapis/python-bigquery/blob/master/samples/tests/test_update_dataset_access.py#L23 so this is just a simple assertion for print.
Also find same kind of assertion in many other samples.

mf2199 commented on storage/cloud-client/snippets_test.py:
```diff
@@ -249,3 +260,30 @@ def test_copy_blob(test_blob):
+
+    bucket = storage_disable_versioning.disable_versioning(test_bucket)
+    out, _ = capsys.readouterr()
+    assert "Versioning was disabled for bucket" in out
```
Same here and henceforth.

mf2199 commented on storage/cloud-client/storage_disable_bucket_lifecycle_management.py:
```diff
@@ -0,0 +1,41 @@
+    bucket.patch()
+    rules = bucket.lifecycle_rules
+
+    print("Lifecycle management is disable for bucket {} and rule is {}".format(bucket_name, list(rules)))
```
"Lifecycle management is disabled for bucket '{}', and the rules are: {}"

mf2199 commented on storage/cloud-client/storage_enable_bucket_lifecycle_management.py:
```diff
@@ -0,0 +1,45 @@
+    bucket = storage_client.get_bucket(bucket_name)
+    rules = bucket.lifecycle_rules
+
+    print("Lifecycle management for bucket {} is {}".format(bucket_name, list(rules)))
```
"Lifecycle management rules for bucket '{}' are: {}"

mf2199 replied:
"Lifecycle management rules for bucket '{}' are: {}"

mf2199 commented on storage/cloud-client/storage_enable_bucket_lifecycle_management.py:
```diff
@@ -0,0 +1,45 @@
+    bucket.patch()
+
+    rules = bucket.lifecycle_rules
+    print("Lifecycle management is enable for bucket {} and rule is {}".format(bucket_name, list(rules)))
```
See the previous comments.

mf2199 commented on storage/cloud-client/storage_set_metadata.py:
```diff
@@ -0,0 +1,40 @@
+    metadata = {'color': 'Red', 'name': 'Test'}
+    blob.metadata = metadata
+
+    print("The metadata of a blob {} is {}".format(blob.name, blob.metadata))
```
"The metadata for the blob '{}' is: {}"

mf2199 reviewed (commented):
Looks good overall.

HemangChothani committed:
docs(storage): nits

HemangChothani committed:
docs(storage): lint fix

HemangChothani committed:
Merge branch 'master' of https://github.com/GoogleCloudPlatform/python-docs-samples into storage_sample_docs_1

renovate-bot committed:
chore(deps): update dependency twilio to v6.39.0 (#3585)

renovate-bot committed:
chore(deps): update dependency sendgrid to v6.3.0 (#3584)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
| [sendgrid](https://togithub.com/sendgrid/sendgrid-python) | minor | `==6.2.2` -> `==6.3.0` |

---

### Release Notes

<details>
<summary>sendgrid/sendgrid-python</summary>

### [`v6.3.0`](https://togithub.com/sendgrid/sendgrid-python/blob/master/CHANGELOG.md#&#8203;2020-04-29-Version-630)

[Compare Source](https://togithub.com/sendgrid/sendgrid-python/compare/6.2.2...6.3.0)

**Library - Feature**

-   [PR #&#8203;882](https://togithub.com/sendgrid/sendgrid-python/pull/882): add support for Twilio Email. Thanks to [@&#8203;childish-sambino](https://togithub.com/childish-sambino)!

</details>

---

### Renovate configuration

:date: **Schedule**: At any time (no schedule defined).

:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

:recycle: **Rebasing**: Never, or you tick the rebase/retry checkbox.

:no_bell: **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#GoogleCloudPlatform/python-docs-samples).

renovate-bot committed:
chore(deps): update dependency redis to v3.5.0 (#3583)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
| [redis](https://togithub.com/andymccurdy/redis-py) | minor | `==3.4.1` -> `==3.5.0` |

---

### Release Notes

<details>
<summary>andymccurdy/redis-py</summary>

### [`v3.5.0`](https://togithub.com/andymccurdy/redis-py/compare/3.4.1...3.5.0)

[Compare Source](https://togithub.com/andymccurdy/redis-py/compare/3.4.1...3.5.0)

</details>

---

### Renovate configuration

:date: **Schedule**: At any time (no schedule defined).

:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

:recycle: **Rebasing**: Never, or you tick the rebase/retry checkbox.

:no_bell: **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#GoogleCloudPlatform/python-docs-samples).

Commit:
[storage] fix: use a different bucket for requester_pays_test (#3655)

* [storage] fix: use a different bucket for requester_pays_test

fixes #3654

* rename to README.md, added the envvar to the template

* add REQUESTER_PAYS_TEST_BUCKET env var

* just use REQUESTER_PAYS_TEST_BUCKET

munkhuushmgl committed:
translate: fix glossary leak issue (#3572)

* fix glossary leak issue

* removed try/catch from teardown methods, removed sample_ prefix from all other methods

* added specific exceptions to tests, added backoff tags to tests

* fixed the lint issues

* reordered imports

* moved backoff inside methd and removed Retry

* corrected import nit

dinagraves committed:
Markdown tutorial (#3547)

* Default templates load properly

* Fully integrated markdown service

* Adding tests

* End to end tests

* Improving the tests and adding region tags

* Renaming tests

* Moving e2e tests

* Removing unused dependency

* Adding empty requirements.txt for test driver

* Sanitizing input

* Added test for missing url

* Reformatting the yaml and sanitizing markdown html

* Removing pytest from requirements.txt

* correctly tearing down

* add a noxfile for e2e test

* Creating unique service names for every test run

* use uuid

* activate service account

* set gcloud project

Co-authored-by: Takashi Matsuo <tmatsuo@google.com>

billyjacobson committed:
bigtable: Handle dev instances and use storage utilization in metric scaler (#3119)

* WIP handle development instances in metric scaler

* use storage utilization and tested

* Fix metric queries

* remove tests for low storage util

* cleanup metric query

* EOF new line

* use uuid instead of random

* lint

* fix uuid length

* fix uuid length

* fix uuid length (again)

Co-authored-by: Christopher Wilcox <crwilcox@google.com>
Co-authored-by: Takashi Matsuo <tmatsuo@google.com>

renovate-bot committed:
chore(deps): update dependency boto3 to v1.13.0 (#3582)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
| [boto3](https://togithub.com/boto/boto3) | minor | `==1.12.48` -> `==1.13.0` |

---

### Release Notes

<details>
<summary>boto/boto3</summary>

### [`v1.13.0`](https://togithub.com/boto/boto3/blob/master/CHANGELOG.rst#&#8203;1130)

[Compare Source](https://togithub.com/boto/boto3/compare/1.12.49...1.13.0)

======

-   api-change:`schemas`: [``botocore``] Update schemas client to latest version
-   api-change:`iot`: [``botocore``] Update iot client to latest version
-   api-change:`lambda`: [``botocore``] Update lambda client to latest version
-   api-change:`storagegateway`: [``botocore``] Update storagegateway client to latest version
-   api-change:`iotevents`: [``botocore``] Update iotevents client to latest version
-   feature:Exceptions: [``botocore``] Added support for parsing modeled exception fields.
-   api-change:`mediaconvert`: [``botocore``] Update mediaconvert client to latest version

### [`v1.12.49`](https://togithub.com/boto/boto3/blob/master/CHANGELOG.rst#&#8203;11249)

[Compare Source](https://togithub.com/boto/boto3/compare/1.12.48...1.12.49)

=======

-   api-change:`iotsitewise`: [``botocore``] Update iotsitewise client to latest version
-   api-change:`waf`: [``botocore``] Update waf client to latest version
-   api-change:`waf-regional`: [``botocore``] Update waf-regional client to latest version
-   api-change:`transcribe`: [``botocore``] Update transcribe client to latest version
-   api-change:`servicediscovery`: [``botocore``] Update servicediscovery client to latest version

</details>

---

### Renovate configuration

:date: **Schedule**: At any time (no schedule defined).

:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

:recycle: **Rebasing**: Never, or you tick the rebase/retry checkbox.

:no_bell: **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#GoogleCloudPlatform/python-docs-samples).

bkuang committed:
Add kms/attestations/ directory with sample script to verify attestations (#3262)

Co-authored-by: Benson Kuang <benkuang@google.com>
Co-authored-by: Takashi Matsuo <tmatsuo@google.com>

renovate-bot committed:
chore(deps): update dependency bleach to v3.1.5 (#3657)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
| [bleach](https://togithub.com/mozilla/bleach) | patch | `==3.1.4` -> `==3.1.5` |

---

### Release Notes

<details>
<summary>mozilla/bleach</summary>

### [`v3.1.5`](https://togithub.com/mozilla/bleach/blob/master/CHANGES#Version-315-April-29th-2020)

[Compare Source](https://togithub.com/mozilla/bleach/compare/v3.1.4...v3.1.5)

**Security fixes**

None

**Features**

None

**Bug fixes**

-   replace missing `setuptools` dependency with `packaging`. Thank you Benjamin Peterson.

</details>

---

### Renovate configuration

:date: **Schedule**: At any time (no schedule defined).

:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

:recycle: **Rebasing**: Never, or you tick the rebase/retry checkbox.

:no_bell: **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#GoogleCloudPlatform/python-docs-samples).

grayside committed:
run: Dockerfile COPY to ./ (#3665)

shollyman committed:
fix: bump ipython version to address compilation issues with py3.8 (#3667)

Fixes: #3658
Fixes: #3659
Fixes: #3660

Co-authored-by: Takashi Matsuo <tmatsuo@google.com>

leahecole committed:
Merge branch 'master' into storage_sample_docs_1

HemangChothani commented:
Merged.
